hikari/impl/cache.py

```diff
@@ -1473,8 +1473,13 @@ def _garbage_collect_message(
             if guild_record:
                 self._garbage_collect_member(guild_record, message.object.member, decrement=1)
 
-        if message.object.referenced_message:
-            self._garbage_collect_message(message.object.referenced_message, decrement=1)
+        referenced_message = message.object.referenced_message
+        if not referenced_message and message.object.message_reference and message.object.message_reference.id:
+            msg_id = message.object.message_reference.id
+            referenced_message = self._message_entries.get(msg_id) or self._referenced_messages.get(msg_id)
```

Comment:
Are you sure this actually fixes the issue, if a referenced message is cached when message.referenced_message is None then wouldn't that infer there's a case the update/set logic isn't accounting for (possibly to do with the idk lets say the referenced message being deleted and that not cascading to places where its being referenced).

Reply:
I'm not aware if there are any other cases, but another way that I could think of to handle this particular case is to ensure that we're not overriding referenced_message's referenced_message since Discord gives us None.